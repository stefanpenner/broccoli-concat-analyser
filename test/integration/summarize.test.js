'use strict';

const tmp = require('tmp');
const path = require('path');
const copyFixtures = require('../helpers/copy-fixtures');
const fs = require('fs-extra');

const inputFixturePath = 'test/fixtures/input';

const summarize = require('../../lib/summarize');
const summarizeAll = require('../../lib/summarize-all');

function file(dir) {
  if (!fs.existsSync(dir)) {
    return false;
  }
  return fs.readFileSync(dir, 'utf-8');
} 

describe('summarize', function() {
  let tmpPath;

  beforeEach(function() {
    tmpPath = tmp.dirSync().name;
  });

  describe('summarize', function() {

    it('calculates sizes', function() {
      let inputFile = '1-test-app.js';
      let outputFile = '1-test-app.js.out.json';

      copyFixtures(inputFile, inputFixturePath, tmpPath);
      return summarize(path.join(tmpPath, `${inputFile}.json`))
        .then(() => expect(file(path.join(tmpPath, outputFile))).toMatchSnapshot());
    });

    it('rejects promise on minification errors', function() {
      let inputFile = '1-test-app.js';

      copyFixtures(inputFile, inputFixturePath, tmpPath);
      // this will cause terser to fail
      fs.writeFileSync(path.join(tmpPath, inputFile, 'test-app/app.js'), 'foo(;', { flag: 'a' });

      let promise = summarize(path.join(tmpPath, `${inputFile}.json`));
      return expect(promise).rejects.toThrow(/Unexpected token/);
    });

  });

  describe('summarize-all', function() {

    it('calculates sizes of all files', function() {
      let inputFiles = ['1-test-app.js', '3-vendor.css', '8-test-support.css'];

      copyFixtures(inputFiles, inputFixturePath, tmpPath);
      return summarizeAll(tmpPath)
        .then(() => {
          inputFiles.forEach((inputFile) => {
            let outputFile = `${inputFile}.out.json`;
            expect(file(path.join(tmpPath, outputFile))).toMatchSnapshot();
          });
        });
    });

    it('can ignore files', function() {
      let inputFiles = ['1-test-app.js', '3-vendor.css', '8-test-support.css'];
      let ignore = 'test-support.css';
      copyFixtures(inputFiles, inputFixturePath, tmpPath);
      return summarizeAll(tmpPath, ignore)
        .then(() => {
          inputFiles.forEach((inputFile) => {
            let outputFile = `${inputFile}.out.json`;
            if (inputFile === '8-test-support.css') {
              expect(file(path.join(tmpPath, outputFile))).not.toBeTruthy();
            } else {
              expect(file(path.join(tmpPath, outputFile))).toMatchSnapshot();
            }
          });
        });
    });

  });
});

