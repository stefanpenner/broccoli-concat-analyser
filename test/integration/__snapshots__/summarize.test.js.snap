// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summarize summarize calculates sizes 1`] = `
"{
  \\"outputFile\\": \\"/assets/test-app.js\\",
  \\"files\\": [
    {
      \\"relativePath\\": \\"test-app/app.js\\",
      \\"sizes\\": {
        \\"raw\\": 606,
        \\"uglified\\": 375,
        \\"compressed\\": 119.82860520094562
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/app-version.js\\",
      \\"sizes\\": {
        \\"raw\\": 510,
        \\"uglified\\": 311,
        \\"compressed\\": 99.37785657998424
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/container-debug-adapter.js\\",
      \\"sizes\\": {
        \\"raw\\": 590,
        \\"uglified\\": 428,
        \\"compressed\\": 136.76438140267928
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/export-application-global.js\\",
      \\"sizes\\": {
        \\"raw\\": 1377,
        \\"uglified\\": 684,
        \\"compressed\\": 218.56737588652481
      }
    },
    {
      \\"relativePath\\": \\"test-app/resolver.js\\",
      \\"sizes\\": {
        \\"raw\\": 231,
        \\"uglified\\": 154,
        \\"compressed\\": 49.20961386918834
      }
    },
    {
      \\"relativePath\\": \\"test-app/router.js\\",
      \\"sizes\\": {
        \\"raw\\": 425,
        \\"uglified\\": 264,
        \\"compressed\\": 84.35933806146572
      }
    },
    {
      \\"relativePath\\": \\"test-app/templates/application.js\\",
      \\"sizes\\": {
        \\"raw\\": 399,
        \\"uglified\\": 322,
        \\"compressed\\": 102.89282899921197
      }
    }
  ]
}"
`;

exports[`summarize summarize-all calculates sizes of all files 1`] = `
"{
  \\"outputFile\\": \\"/assets/test-app.js\\",
  \\"files\\": [
    {
      \\"relativePath\\": \\"test-app/app.js\\",
      \\"sizes\\": {
        \\"raw\\": 606,
        \\"uglified\\": 375,
        \\"compressed\\": 119.82860520094562
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/app-version.js\\",
      \\"sizes\\": {
        \\"raw\\": 510,
        \\"uglified\\": 311,
        \\"compressed\\": 99.37785657998424
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/container-debug-adapter.js\\",
      \\"sizes\\": {
        \\"raw\\": 590,
        \\"uglified\\": 428,
        \\"compressed\\": 136.76438140267928
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/export-application-global.js\\",
      \\"sizes\\": {
        \\"raw\\": 1377,
        \\"uglified\\": 684,
        \\"compressed\\": 218.56737588652481
      }
    },
    {
      \\"relativePath\\": \\"test-app/resolver.js\\",
      \\"sizes\\": {
        \\"raw\\": 231,
        \\"uglified\\": 154,
        \\"compressed\\": 49.20961386918834
      }
    },
    {
      \\"relativePath\\": \\"test-app/router.js\\",
      \\"sizes\\": {
        \\"raw\\": 425,
        \\"uglified\\": 264,
        \\"compressed\\": 84.35933806146572
      }
    },
    {
      \\"relativePath\\": \\"test-app/templates/application.js\\",
      \\"sizes\\": {
        \\"raw\\": 399,
        \\"uglified\\": 322,
        \\"compressed\\": 102.89282899921197
      }
    }
  ]
}"
`;

exports[`summarize summarize-all calculates sizes of all files 2`] = `
"{
  \\"outputFile\\": \\"/assets/vendor.css\\",
  \\"files\\": []
}"
`;

exports[`summarize summarize-all calculates sizes of all files 3`] = `
"{
  \\"outputFile\\": \\"/assets/test-support.css\\",
  \\"files\\": [
    {
      \\"relativePath\\": \\"vendor/qunit/qunit.css\\",
      \\"sizes\\": {
        \\"raw\\": 7875,
        \\"compressed\\": 2123.0700712589073
      }
    },
    {
      \\"relativePath\\": \\"vendor/ember-qunit/test-container-styles.css\\",
      \\"sizes\\": {
        \\"raw\\": 544,
        \\"compressed\\": 146.6603325415677
      }
    }
  ]
}"
`;

exports[`summarize summarize-all can ignore files 1`] = `
"{
  \\"outputFile\\": \\"/assets/test-app.js\\",
  \\"files\\": [
    {
      \\"relativePath\\": \\"test-app/app.js\\",
      \\"sizes\\": {
        \\"raw\\": 606,
        \\"uglified\\": 375,
        \\"compressed\\": 119.82860520094562
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/app-version.js\\",
      \\"sizes\\": {
        \\"raw\\": 510,
        \\"uglified\\": 311,
        \\"compressed\\": 99.37785657998424
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/container-debug-adapter.js\\",
      \\"sizes\\": {
        \\"raw\\": 590,
        \\"uglified\\": 428,
        \\"compressed\\": 136.76438140267928
      }
    },
    {
      \\"relativePath\\": \\"test-app/initializers/export-application-global.js\\",
      \\"sizes\\": {
        \\"raw\\": 1377,
        \\"uglified\\": 684,
        \\"compressed\\": 218.56737588652481
      }
    },
    {
      \\"relativePath\\": \\"test-app/resolver.js\\",
      \\"sizes\\": {
        \\"raw\\": 231,
        \\"uglified\\": 154,
        \\"compressed\\": 49.20961386918834
      }
    },
    {
      \\"relativePath\\": \\"test-app/router.js\\",
      \\"sizes\\": {
        \\"raw\\": 425,
        \\"uglified\\": 264,
        \\"compressed\\": 84.35933806146572
      }
    },
    {
      \\"relativePath\\": \\"test-app/templates/application.js\\",
      \\"sizes\\": {
        \\"raw\\": 399,
        \\"uglified\\": 322,
        \\"compressed\\": 102.89282899921197
      }
    }
  ]
}"
`;

exports[`summarize summarize-all can ignore files 2`] = `
"{
  \\"outputFile\\": \\"/assets/vendor.css\\",
  \\"files\\": []
}"
`;
